### DISCUSSION TITLE
Sort or filter search results based on date created or updated?

### DISCUSSION THREAD

**Discussion by jfdrew** [zim-desktop-wiki/zim-desktop-wiki#1965] [Q&A]
Let me start by saying "thank you" for creating Zim.  I find Zim valuable, and I enjoy using it.  

Now, on to my request...I am looking for documentation or suggestions on how to find notes based on date-of-creation or date-of-last-modification.

For example, suppose I Search for notes about "Fred".  Zim shows notes based on a relevance score.  This works well and is often sufficient.  

But sometimes I would like to be able to sort or filter the result set based on date-of-creation or date-of-last-modification:
* Notes that include "Fred", sorted by most recent update, not relevance. 
* Notes that include "Fred", created between 2022-01-15 and 2022-02-15

Questions:
a) Is this possible in existing GUI (and I just have not found it)?
b) Is there a plugin?
c) Is there a way to do this with Zim command line?
d) Is there a way to do this with some sort of scripting (shell, python, other?) 

Context:
* I am running Zim 74.2 on a Mac
* I am coming to Zim from Evernote.  I love Zim's wiki-ness (as opposed to Evernote's list centric ui), but I do miss the ability to optionally search and sort based on dates.

**Comment by jaap-karssenberg** [comment]
Updated timestamp is in the index, so reasonable easy to add it to the
search query syntax.

When also adding a column for this is the search view we can remove the
"recent changes" dialog and just use search for that.


Op wo 2 mrt. 2022 01:57 schreef jfdrew ***@***.***>:

**Comment by introt** [comment]
> d) Is there a way to do this with some sort of scripting (shell, python, other?)

I was going to say yes, but then I remembered the way Zim saves files resets the creation date. However, if you're feeling adventurous, you can certainly try parsing the dates in the templates: https://stackoverflow.com/questions/60835418/bash-find-all-files-based-on-date-range-in-file-names

For modification dates, you can just chain some standard UNIX utilities. Here's a dirty one-liner example, combining [an earlier answer of mine](https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1735#discussioncomment-1613691) with this [StackOverFlow post](https://stackoverflow.com/questions/18339307/find-files-in-created-between-a-date-range).

`find $NOTEBOOK_DIR -name '*.txt' -newermt "2022-01-15 00:00:00" ! -newermt "2022-02-16 00:00:00" -exec grep -l -m 1 '^Content-Type: text/x-zim-wiki$' {} + |  xargs ls -1t | xargs grep "Fred"`

`-newermt` means "newer modification time than", and `! -newermt` the opposite; note the off-by-one date in the latter. Add a `-l` to the final grep if you want to list the filenames and not see the text matches, and(/or, though that could get messy) `sed -e 's%/%:%g' -e 's/.txt//g'` after that to get valid page names instead of filenames for quick copy & paste linking.
